ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/jta/TransactionManagerService.java

```diff
@@ -56,6 +57,16 @@ public void stop()
     {
     }
 
+    /**
+     * This will need to be called from WildFly ServerActivity in a manner similar to how RecoverySuspendController works
+     */
+    public void suspend() {
```

Comment:
Maybe this method is not actually directly https://issues.redhat.com/browse/JBTM-3894 and should be considered a separate JBTM to provide a method to suspend the transaction manager?

Or we rename JBTM-3894 to be `Introduce the ability to suspend the transaction manager and block the suspension of the Recovery Manager until all transactions are recovered`

Reply:
Yeah, JBTM-3894 doesn't cover this modification. Maybe JBTM-3893 does? I mean, it might become a subtask of JBTM-3893 when it comes to implement the new `SuspendBlockingRecoveryModule` into JTA/JTS

Reply:
I added this code to JBTM-3893. Thanks Tom